### PR TITLE
resync on service update

### PR DESCRIPTION
**Pull request by ilackarms**
# Description

Resyncs the ggv2 controller when a k8s service is updated

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/6332

### CODE REVIEW

File: test/kubernetes/e2e/features/services/types.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	proxyService    = &corev1.Service{ObjectMeta: glooProxyObjectMeta}
+
+	// curlPod is the Pod that will be used to execute curl requests, and is defined in the upstream manifest files
+	curlPodExecOpt = kubectl.PodExecOptions{
+		Name:      "curl",
+		Namespace: "curl",
+		Container: "curl",
+	}
+
+	expectedSvcResp = &testmatchers.HttpResponse{
```
Comment (sam-heilbron) [inline test/kubernetes/e2e/features/services/types.go:37]: Can we just use [HaveOkResponse()](https://github.com/solo-io/gloo/blob/main/test/gomega/matchers/have_http_response.go#L19 ) instead?

File: test/kubernetes/e2e/features/services/types.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	// Proxy resource to be translated
+	glooProxyObjectMeta = metav1.ObjectMeta{
+		Name:      "gloo-proxy-gw",
+		Namespace: "default",
+	}
+	proxyDeployment = &appsv1.Deployment{ObjectMeta: glooProxyObjectMeta}
+	proxyService    = &corev1.Service{ObjectMeta: glooProxyObjectMeta}
+
+	// curlPod is the Pod that will be used to execute curl requests, and is defined in the upstream manifest files
+	curlPodExecOpt = kubectl.PodExecOptions{
```
Comment (sam-heilbron) [inline test/kubernetes/e2e/features/services/types.go:31]: Let's use [defaults.CurlPodExecOpt](https://github.com/solo-io/gloo/blob/main/test/kubernetes/e2e/defaults/defaults.go#L11) instead

File: test/kubernetes/e2e/features/services/suite.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		s.NoError(err, "can delete manifest")
+		err = s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, serviceManifest)
+		s.NoError(err, "can delete manifest")
+		s.testInstallation.Assertions.EventuallyObjectsNotExist(s.ctx, proxyService, proxyDeployment)
+	})
+
+	err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, routeWithServiceManifest)
+	s.Assert().NoError(err, "can apply gloo.solo.io Route manifest")
+
+	// apply the sercice manifest separately, after the route table is applied, to ensure it can be applied after the route table
```
Comment (sam-heilbron) [inline test/kubernetes/e2e/features/services/suite.go:42]: is the phrase "route table" accurate here?

I do like this comment in general clarifying what behavior we're testing

**Review by sam-heilbron — COMMENTED** [review]
a few small nits on the tests, plus I think well need a changelog

File: test/kubernetes/e2e/features/services/types.go (hunk 1 above)
Comment (ilackarms) [reply test/kubernetes/e2e/features/services/types.go:37]: i'm not sure how to use that with our test framework. do you have an example of this in use?

File: test/kubernetes/e2e/features/services/suite.go (hunk 3 above)
Comment (ilackarms) [reply test/kubernetes/e2e/features/services/suite.go:42]: i consider HTTRoute as route table and still use the terms interchangeably. "Route" is extremely vague and confusing in this usage; i try to use "Route" to refer to individual route rules in a route table. I think the distinction is helpful even though the CRD itself is not called route table

File: test/kubernetes/e2e/features/services/suite.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"context"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/solo-io/gloo/pkg/utils/requestutils/curl"
+	"github.com/solo-io/gloo/test/kubernetes/e2e"
+)
+
+// testingSuite is the entire Suite of tests for the "Upstream" feature
```
Comment (lgadban) [inline test/kubernetes/e2e/features/services/suite.go:12]: this isn't "Upstreams" right?

File: test/kubernetes/e2e/features/services/suite.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func NewTestingSuite(ctx context.Context, testInst *e2e.TestInstallation) suite.TestingSuite {
+	return &testingSuite{
+		ctx:              ctx,
+		testInstallation: testInst,
+	}
+}
+
+func (s *testingSuite) TestConfigureBackingDestinationsWithUpstream() {
```
Comment (lgadban) [inline test/kubernetes/e2e/features/services/suite.go:30]: same here, should be ...DestinationsWithKubeService or something

**Review by lgadban — COMMENTED** [review]
Found some small naming things, also I think the feature suites need to be added to tests to actually be run

File: test/kubernetes/e2e/features/services/types.go (hunk 1 above)
Comment (jmhbh) [reply test/kubernetes/e2e/features/services/types.go:33]: found one here - https://github.com/solo-io/gloo/blob/main/test/e2e/ratelimit_test.go#L308

**Review by jmhbh — DISMISSED** [review]
LGTM pending Lawrence and Sam's comments being resolved

File: test/kubernetes/e2e/features/services/types.go (hunk 1 above)
Comment (sam-heilbron) [reply test/kubernetes/e2e/features/services/types.go:33]: Yeah this is purely a testing utility, so our framework doesn't have strong opinions on doing/not doing this (ie it should be possible to use whatever utilities you need)

**Review by sam-heilbron — DISMISSED** [review]

File: changelog/v1.17.0-rc3/ggv2-resync-on-svc.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    issueLink: https://github.com/solo-io/solo-projects/issues/5890
+    resolvesIssue: false
+    description: |
+      1. Add API for running postRouting transformations. This means that the transformation filter
+      will be run as a part of the router filter, rather than during the standard HTTP filter chain.
+      This allows access to endpoint specific information which is not available during the standard
+      fiter chain.
+      2. Add API for adding metadata to endpoints in static/failover upstreams. This metadata can
+      then be accessed in the transformation filter mentioned above.
+      3. Add new plugin type to allow adding filter in the upstream filter chain.
```
Comment (lgadban) [inline changelog/v1.17.0-rc3/ggv2-resync-on-svc.yaml:12]: This doesn't seem right?

File: changelog/v1.17.0-rc4/ggv2-resync-on-svc.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+changelog:
+  - type: NEW_FEATURE
+    issueLink: https://github.com/solo-io/solo-projects/issues/5890
```
Comment (sam-heilbron) [inline changelog/v1.17.0-rc4/ggv2-resync-on-svc.yaml:3]: I think this is the wrong issue referenced

File: changelog/v1.17.0-rc4/ggv2-resync-on-svc.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    resolvesIssue: false
+    description: |
+      1. Add API for running postRouting transformations. This means that the transformation filter
```
Comment (sam-heilbron) [inline changelog/v1.17.0-rc4/ggv2-resync-on-svc.yaml:6]: I don't think this is accurate

File: changelog/v1.17.0-rc4/ggv2-resync-on-svc.yaml (hunk 7 above)
Comment (sam-heilbron) [reply changelog/v1.17.0-rc4/ggv2-resync-on-svc.yaml:3]:
```suggestion
    issueLink: https://github.com/solo-io/solo-projects/issues/6332
```

**Review by sam-heilbron — APPROVED** [review]

**Review by lgadban — APPROVED** [review]

**Pull request merged**